### PR TITLE
added subscribe

### PR DESCRIPTION
**Pull request by betzrhodes**

### CODE REVIEW

File: TheThingsAPI.class.nut [hunk 1]
```diff
@@ -77,15 +73,34 @@ class TheThingsAPI {
...
+    //              data: the decoded data from
+    function subscribe(cb = null) {
+        // make sure only one subscribe request is active per thing
+        if (_subscribeRequest != null) _subscribeRequest.cancel();
+
+        // create new subscribe request
+        _subscribeRequest = http.get(format("%s%s", URLROOT, _token), HEADERS_GET);
+
+        // open subscribe long polling request
+        _subscribeRequest.sendasync(_subscribeResponseFactory(cb), _subscribeOnDataFactory(cb), 60000);
```
Comment (ppetrosh) [inline TheThingsAPI.class.nut]: should 60000 be a constant or a configuration option?

File: TheThingsAPI.class.nut [hunk 2]
```diff
@@ -173,9 +223,17 @@ class TheThingsAPI {
         return function(resp) {
             // If we didn't get a 2xx status code, there was an error..
             if (resp.statuscode < 200 || resp.statuscode >= 300) {
+                if (resp.statuscode == 429) {
```
Comment (ppetrosh) [inline TheThingsAPI.class.nut]: Just curious, why do we specially handle this code?

**Comment by ppetrosh** [comment]
Approved from my side.

File: TheThingsAPI.class.nut (hunk 2 above)
Comment (betzrhodes) [reply TheThingsAPI.class.nut]: The 429 was originally getting overwritten with a JSON decoding error. Looks like I fixed the JSON issue, so I will remove the special handling for the 429.

**Pull request merged**